### PR TITLE
fix: dedup MCP AgentOS config databases and guard get_db against no database

### PR DESCRIPTION
**Pull request by harshsinha03**
## Summary

Small correctness fixes to the AgentOS MCP interface (v2) and the shared `get_db` helper it relies on. The happy path is unchanged; this covers two edge cases plus three doc/cleanup items.

- **`get_agentos_config` lists each database once.** A database shared by several components is registered once per component in `os.dbs`, so the tool returned duplicate ids (e.g. `["pg", "pg", "pg"]`). It now collects the ids into a set — the same expression the REST `/config` route already uses in `os/router.py` — so the MCP config tool and the REST route agree.
- **`get_db` returns a clear `400` when no database is configured.** With no database, `os.dbs` is empty and the old `next(...)` raised a bare `StopIteration`, which inside an `async` function surfaces as the opaque `RuntimeError: coroutine raised StopIteration`. It now raises `HTTPException(400, "No database is configured on this AgentOS")`. This also covers an id mapped to an empty list, and drops a loop variable that shadowed the `dbs` parameter.
- **Import `ToolResult` from the stable `fastmcp.tools` path** instead of the `fastmcp.tools.tool` compatibility shim (which fastmcp plans to remove), matching the existing `from fastmcp.tools import Tool` import in the same module.
- **Docs:** correct a stale `MCPBuiltinTag` comment that still listed the removed `memory` tag, and update the custom-MCP-tool cookbook's built-in count (~19 to 8).

## Type of change

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [x] Improvement
- [ ] Model update
- [ ] Other:

---

## Checklist

- [x] Code complies with style guidelines
- [x] Ran format/validation scripts (`./scripts/format.sh` and `./scripts/validate.sh`)
- [x] Self-review completed
- [x] Documentation updated (comments, docstrings)
- [x] Examples and guides: Relevant cookbook examples have been included or updated (if applicable)
- [ ] Tested in clean environment
- [ ] Tests added/updated (if applicable)

### Duplicate and AI-Generated PR Check

- [x] I have searched existing open pull requests and confirmed that no other PR already addresses this issue
- [ ] If a similar PR exists, I have explained below why this PR is a better approach
- [ ] Check if this PR was entirely AI-generated (by Copilot, Claude Code, Cursor, etc.)

---

## Additional Notes

Verified against a real FastMCP client (in-memory and over HTTP) with a real Postgres:

- No database configured -> the MCP session tools and the REST session/memory routes return a clear `400` instead of the opaque `coroutine raised StopIteration`.
- One database shared by three components -> `get_agentos_config` returns `["pg"]`, matching REST `/config`.

The full MCP test surface was run for regressions (12 files, 402 tests) with no failures, and `./scripts/validate.sh` (ruff + whole-repo mypy) passes. No new unit test was added for the two edge cases; they are exercised by the existing MCP suite plus the manual real-client verification above.

### CODE REVIEW

File: cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 AgentOS app that exposes ONE custom MCP tool routed through an agent, with the
 built-in MCP tools disabled and the server gated to its owner.
 
-This is the "one tool" shape: instead of the ~19 built-in AgentOS tools, the MCP
+This is the "one tool" shape: instead of the 8 built-in AgentOS tools, the MCP
```
Comment (ysolanky) [inline cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py:5]: But there are 19 tools now?

File: libs/agno/agno/os/utils.py [hunk 2]
```diff
@@ -311,8 +311,13 @@ async def _has_table(db: Union[BaseDb, AsyncBaseDb, RemoteDb], table_name: str)
             status_code=400, detail="The db_id query parameter is required when using multiple databases"
         )
 
+    # Raise if no database is registered (an empty dict, or ids mapped to empty lists)
+    all_dbs = [db for db_list in dbs.values() for db in db_list]
+    if not all_dbs:
+        raise HTTPException(status_code=400, detail="No database is configured on this AgentOS")
```
Comment (ysolanky) [inline libs/agno/agno/os/utils.py:317]: Raising here is new behavior. Why do we need to raise here?

**Review by kausmeows — APPROVED** [review]

File: cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py (hunk 1 above)
Comment (kausmeows) [reply cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py:5]: no, there are 8 😅

File: libs/agno/agno/os/utils.py (hunk 2 above)
Comment (kausmeows) [reply libs/agno/agno/os/utils.py:317]: actually true, this will create problems for other flows where db strictly is not need on the os right

File: cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py (hunk 1 above)
Comment (harshsinha03) [reply cookbook/05_agent_os/mcp_demo/custom_mcp_tool_example.py:5]: yes

File: libs/agno/agno/os/utils.py (hunk 2 above)
Comment (harshsinha03) [reply libs/agno/agno/os/utils.py:317]: All get_db callers already require a db run flows never call it, and no-db already crashed here before, with this fix it makes that same failure a clean 400 instead of a RuntimeError.

**Review by ysolanky — APPROVED** [review]
